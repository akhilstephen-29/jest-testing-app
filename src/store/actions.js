export function addCounter(payload) {
  console.log(payload)
  return { type: "ADD_COUNTER", payload}
}

export function getPosts(payload) {
  return {type: "GET_POSTS", payload}
}