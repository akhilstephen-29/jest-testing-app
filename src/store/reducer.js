
import { ADD_COUNTER, GET_POSTS } from "../utils/constants";
import {addCounter, getPosts} from "./actions.js";

const initialState ={
  counter : 0,
  posts: {}
}

const reducer = (state = initialState, action ) => {
  console.log(action.payload)
  switch(action.type){
    case ADD_COUNTER:
      return Object.assign({}, state, {counter: action.payload})
    case GET_POSTS:
      return Object.assign({}, state, {posts: action.payload})
    default:  return state;
  }

}

export default reducer;

export function mapStateToProps(state) {
  return state;
}

export function mapDispatchToProps(dispatch){
  return {
    addCounter: counter => dispatch(addCounter(counter)),
    getPosts: posts => dispatch(getPosts(posts))
  }
}