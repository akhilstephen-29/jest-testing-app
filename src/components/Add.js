import React from "react";
import { connect } from "react-redux";
import * as service from "../utils/fetchAPI";

import {mapStateToProps, mapDispatchToProps} from "../store/reducer";
class Add extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            response: {}
        };
        this.submitClick = this.submitClick.bind(this);
        this.axiosSuccess = this.axiosSuccess.bind(this);
    }
    axiosSuccess(res){
        this.setState({response: res.data})
        console.log(this.state.response)
        this.props.getPosts(res.data.data);
    }
    submitClick(){
        this.props.addCounter(10);
        let input = {
            url: "https://reqres.in/api/users?page=2",
            callbackSuccess: this.axiosSuccess
        }
        if(process.REDUX_MOCK_ENABLED){
            
        }
        service.fetchGETAPI(input);
    }
    render() {
        return (
            <div className="App">
                <h1>Add Function</h1>
                <h4>{this.props.counter}</h4>
                <button onClick={this.submitClick}>Submit</button>
            </div>
        )
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(Add);