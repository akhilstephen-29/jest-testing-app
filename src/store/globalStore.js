import {createStore} from "redux";
import reducer from "../store/reducer";

export default globalStore = createStore(reducer);