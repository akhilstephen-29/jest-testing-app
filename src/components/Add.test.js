import Add from "./Add";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import {addCounter} from "../store/actions";
import toJson from "enzyme-to-json";
import renderer from "react-test-renderer";

const mockStore = configureStore([]);

describe("Add Functions", () => {
    let store;
    let wrapper;
    
    
    beforeEach(() => {
        store = mockStore({
            counter: 5
        })

        store.dispatch = jest.fn();
        wrapper = shallow(<Add store={store} />).childAt(0).dive();
    });
    test("[SNAPSHOT TEST ADD]", () => {
        let snapshotWrapper = renderer.create(<Add store={store} />);
        let tree = snapshotWrapper.toJSON();
        expect(tree).toMatchSnapshot();
        
    })
    test("Sample est", () => {
        expect(1).toBe(1);
    })
    test("simulate functions", () => {
        let classInstance = wrapper.instance();
        classInstance.submitClick();
        wrapper.update();
        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(addCounter(10));
    })
})