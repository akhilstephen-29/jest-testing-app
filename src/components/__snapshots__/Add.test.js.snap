// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Functions [SNAPSHOT TEST ADD] 1`] = `
<div
  className="App"
>
  <h1>
    Add Function
  </h1>
  <h4>
    5
  </h4>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
